### PR TITLE
WIP: Improve doxygen documentation for the Image class

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

File: Code/Common/include/sitkImage.h [hunk 1]
```diff
@@ -196,10 +239,19 @@ namespace simple
     /** Transform continuous index to physical point */
     std::vector< double > TransformContinuousIndexToPhysicalPoint( const std::vector< double > &index) const;
 
+    /** Get the number of pixels the Image is each dimension as a
```
Comment (dave3d) [inline Code/Common/include/sitkImage.h]: "is **in** each dimension"

**Review by dave3d — APPROVED** [review]
Other than my tiny grammatical fix, looks good to me.

File: Code/Common/include/sitkImage.h [hunk 2]
```diff
@@ -49,7 +49,32 @@ namespace simple
...
-   * \brief The main Image class for SimpleITK
+   * \brief The Image class for SimpleITK
+   *
+   * This Image class can represent 2D, 3D, and 4D images. The pixel
+   * types may be a scalar, a multi-component vector or a
+   * run-length-encoded (RLE) "label". The dimension, pixel type and
+   * size is specified at construction and can only be changed by
+   * assignment.
+   *
+   * A fundamental concept of ITK images are they occupy physical
```
Comment (zivy) [inline Code/Common/include/sitkImage.h]: Rephrase: "A fundamental concept of ITK images is that they occupy..."

File: Code/Common/include/sitkImage.h [hunk 3]
```diff
@@ -49,7 +49,32 @@ namespace simple
   class PimpleImageBase;
 
   /** \class Image
-   * \brief The main Image class for SimpleITK
+   * \brief The Image class for SimpleITK
+   *
+   * This Image class can represent 2D, 3D, and 4D images. The pixel
+   * types may be a scalar, a multi-component vector or a
+   * run-length-encoded (RLE) "label". The dimension, pixel type and
+   * size is specified at construction and can only be changed by
```
Comment (zivy) [inline Code/Common/include/sitkImage.h]: Not sure what the phrase "can only be changed by assignment". I would end the sentence "specified at construction." I don't  conceptually think that assigning another image is a change in size or pixel type.

File: Code/Common/include/sitkImage.h [hunk 4]
```diff
@@ -49,7 +49,32 @@ namespace simple
...
+   * space where the image is defined by an origin, spacing, and
+   * direction cosine matrix. The attributes are taken into
+   * consideration when doing most operations on an image. A meta-data
+   * dictionary is also associated with the image, which may contain
+   * additional fields from reading but these attributes are not
+   * propagated by image filters.
+   *
+   * The SimpleITK Image provides a single facade interface to several
+   * ITK image types. Internally, the SimpleITK Image maintains a
+   * pointer to the ITK image class, and performing reference counting
```
Comment (zivy) [inline Code/Common/include/sitkImage.h]: keep the same tense: "...and performs.."

File: Code/Common/include/sitkImage.h [hunk 5]
```diff
@@ -159,14 +199,17 @@ namespace simple
...
      */
     std::vector< double > GetOrigin( void ) const;
     void SetOrigin( const std::vector< double > &origin );
     /** @} */
 
-    /** Get/Set the Spacing
+    /** Get/Set the Spacing of the Image as an std::vector .
+     *
+     * The spacing describes the physical size of each pixel. The
+     * length of the vector is equal to the dimension of the Image.
```
Comment (zivy) [inline Code/Common/include/sitkImage.h]: The length of the vector can be equal or greater than the dimension of the image? Should the "greater" option be explicitly stated?

**Review by zivy — COMMENTED** [review]
Some minor questions/corrections.

**Pull request merged**